holoviews/core/data/grid.py

```diff
@@ -102,18 +102,79 @@ def length(cls, dataset):
 
 
     @classmethod
+    def get_coords(cls, dataset, dim, ordered=False):
+        """
+        Returns the coordinates along a dimension.
+        """
+        data = dataset.data[dim]
+        if ordered and np.all(data[1:] < data[:-1]):
+            data = data[::-1]
+        return data
+
+
+    @classmethod
+    def expanded_coords(cls, dataset, dim):
+        arrays = [cls.get_coords(dataset, d.name, True)
+                  for d in dataset.kdims]
+        idx = dataset.get_dimension_index(dim)
+        return util.cartesian_product(arrays)[idx]
+
+
+    @classmethod
+    def canonicalize(cls, dataset, data, coord_dims=None):
+        """
+        Canonicalize takes an array of values as input and
+        reorients and transposes it to match the canonical
+        format expected by plotting functions. In addition
```

Comment:
It is probably too much of a pain to stick in the docstring but it would be good to describe our canonical format properly somewhere in our docs. Not something that should hold up this PR though!